### PR TITLE
feat: Move data warehouse sources into pipeline

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Wanted to play with what an alternative could look like for data sources https://github.com/PostHog/posthog/pull/24059

## Changes

Screenshot 2024-08-01 at 11 21 30 AM

* Moves everything into data pipelines
* Simplifies the warehouse stuff into one "sources" area
* Moves the "Database tables" into the sources tab

## TODO
- [x] Fix up remaining paths so we only have one flow for the data warehouse source management
- [x] Agree if this is the right direction

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Comment by EDsCODE** [comment]
I'm good with this format too as long as we address the add-on perception @raquelmsmith.

1. I think we can remove the "all tables" section as this will be in the editor schema
2. We'll need to bound the number of sources for each section (managed,self-managed) if there are 100 managed sources, you won't be able to see self-managed is on this page too

**Comment by benjackwhite** [comment]
> I'm good with this format too as long as we address the add-on perception @raquelmsmith.
> 
> 1. I think we can remove the "all tables" section as this will be in the editor schema
> 2. We'll need to bound the number of sources for each section (managed,self-managed) if there are 100 managed sources, you won't be able to see self-managed is on this page too

1. Sure.
2. tbh if we are hitting that many sources, we likely need better tooling like search etc.

**Comment by EDsCODE** [comment]
Maybe 100 is an exaggeration. I was just looking at the number of sources we have now and it would make stacked tables a bit confusing. Could limit the tables to show 10? and then paginate beyond that

**Comment by raquelmsmith** [comment]
We can rename the addon. I just don't know what we'd rename it _to_ 🤔

**Review by Gilbert09 — APPROVED** [review]

**Comment by corywatilo** [comment]
May feel pedantic but two things I'd like us to consider here:

1. The 3000 nav re-org moved _Data_ and _People_ to the top section along with Dashboards and Notebooks. It was intended to be the section for all your _project-related data_. I could be convinced to separate dashboards and notebooks from the data layer (since they're an abstraction from it), but think it makes sense to keep _Data_ and _People_ together.
2. The term "Events management" is clunky. Think just going with _Events_ does the job, and matches _People_ in brevity.

**Comment by benjackwhite** [comment]
> May feel pedantic but two things I'd like us to consider here:
> 
> 1. The 3000 nav re-org moved _Data_ and _People_ to the top section along with Dashboards and Notebooks. It was intended to be the section for all your _project-related data_. I could be convinced to separate dashboards and notebooks from the data layer (since they're an abstraction from it), but think it makes sense to keep _Data_ and _People_ together.
> 2. The term "Events management" is clunky. Think just going with _Events_ does the job, and matches _People_ in brevity.

This is actually no longer part of this PR so we can agree what to do in follow up

**Comment by corywatilo** [comment]
Not sure if we're still merging this or not but if so...

SQL Studio → SQL studio (casing)

File: frontend/src/scenes/sceneTypes.ts [hunk 1]
```diff
@@ -43,11 +42,9 @@ export enum Scene {
     Surveys = 'Surveys',
     Survey = 'Survey',
     SurveyTemplates = 'SurveyTemplates',
-    DataWarehouse = 'DataWarehouse',
+    DataWarehouse = 'SQL',
```
Comment (Twixes) [inline frontend/src/scenes/sceneTypes.ts:45]: These should stay the same

File: frontend/src/scenes/urls.ts [hunk 2]
```diff
@@ -76,21 +75,28 @@ export const urls = {
         filters?: AnyPartialFilterType,
         dashboardId?: DashboardType['id'] | null,
         query?: string | Record<string, any>
-    ): string =>
-        combineUrl('/insights/new', dashboardId ? { dashboard: dashboardId } : {}, {
+    ): string => {
+        if (query) {
+            return combineUrl(`/sql`, {}, query ? { q: typeof query === 'string' ? query : JSON.stringify(query) } : {})
+                .url
+        }
```
Comment (Twixes) [inline frontend/src/scenes/urls.ts:82]: This isn't right, as insights use `query` too. You can see this in Web analytics, where in this PR "Open as new insight" opens trend insights in… SQL studio. Only queries with `kind: "HogQLQuery"` should be opened (or generally _openable_) in the SQL scene.

**Review by Twixes — DISMISSED** [review]

**Review by Twixes — COMMENTED** [review]
Only looked at the querying/SQL side of things – just a couple of issues before merging

File: frontend/src/scenes/insights/insightSceneLogic.tsx [hunk 3]
```diff
@@ -189,7 +189,7 @@ export const insightSceneLogic = kea<insightSceneLogicType>([
         setSceneState: sharedListeners.reloadInsightLogic,
     })),
     urlToAction(({ actions, values }) => ({
-        '/data-warehouse/*': (_, __, { q }) => {
```
[the remark this answers is not in the document]
Comment (EDsCODE) [reply frontend/src/scenes/insights/insightSceneLogic.tsx:192]: Good point, i set a redirect for all /data-warehouse links to go to /sql

File: frontend/src/scenes/pipeline/sources/Sources.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                <div>
+                    <h2>Managed sources</h2>
+                    <p>
+                        PostHog can connect to external sources and automatically import data from them into the PostHog
+                        data warehouse
+                    </p>
+                    <DataWarehouseManagedSourcesTable />
+                </div>
+                <div>
+                    <h2>Self managed sources</h2>
```
Comment (Twixes) [inline frontend/src/scenes/pipeline/sources/Sources.tsx:40]:
```suggestion
                    <h2>Self-managed sources</h2>
```

**Pull request merged**